### PR TITLE
Wait for build to be enqueued. Use more efficient queue URLs.

### PR DESCRIPTION
**Pull request by rmmh**

### CODE REVIEW

File: prow/cmd/line/main.go [hunk 1]
```diff
@@ -334,12 +334,16 @@ func (c *testClient) TestPRJenkins() error {
 		c.tryCreateStatus("", github.StatusError, "Error starting build.", testInfra)
 		return err
 	}
-	eq, err := c.JenkinsClient.Enqueued(b)
-	if err != nil {
-		c.tryCreateStatus("", github.StatusError, "Error queueing build.", testInfra)
-		return err
+	eq := false
+	for !eq { // Wait for it to appear in the queue
+		time.Sleep(10 * time.Second)
```
Comment (fejta) [inline prow/cmd/line/main.go]: 2-5s seems happier to me. 10s is a long time to wait

Comment (rmmh) [inline prow/cmd/line/main.go]: This doesn't actually delay the build at all, it just takes a little longer to get through the queue.

File: prow/cmd/line/main.go [hunk 2]
```diff
@@ -334,12 +334,16 @@ func (c *testClient) TestPRJenkins() error {
 		c.tryCreateStatus("", github.StatusError, "Error starting build.", testInfra)
 		return err
 	}
-	eq, err := c.JenkinsClient.Enqueued(b)
-	if err != nil {
-		c.tryCreateStatus("", github.StatusError, "Error queueing build.", testInfra)
-		return err
+	eq := false
+	for !eq { // Wait for it to appear in the queue
```
Comment (spxtr) [inline prow/cmd/line/main.go]: If we miss it in the queue then this will loop forever, won't it?

Comment (spxtr) [inline prow/cmd/line/main.go]: No, it won't.

File: prow/jenkins/jenkins.go [hunk 3]
```diff
@@ -163,7 +163,7 @@ func (c *Client) Enqueued(b *Build) (bool, error) {
 	if c.dry {
 		return false, nil
 	}
-	u := fmt.Sprintf("%s/queue/api/json", c.baseURL)
+	u := fmt.Sprintf("%s/%sapi/json", c.baseURL, b.queueURL)
```
Comment (spxtr) [inline prow/jenkins/jenkins.go]: should there be a slash?

**Pull request merged**